libs/partners/openai/langchain_openai/chat_models/base.py

```diff
@@ -135,19 +144,15 @@ def _convert_message_to_dict(message: BaseMessage) -> dict:
             if message_dict["content"] == "":
                 message_dict["content"] = None
     elif isinstance(message, SystemMessage):
-        message_dict = {"role": "system", "content": message.content}
+        message_dict["role"] = "system"
     elif isinstance(message, FunctionMessage):
-        message_dict = {
-            "role": "function",
-            "content": message.content,
-            "name": message.name,
-        }
+        message_dict["role"] = "function"
     elif isinstance(message, ToolMessage):
-        message_dict = {
-            "role": "tool",
-            "content": message.content,
-            "tool_call_id": message.tool_call_id,
-        }
+        message_dict["role"] = "tool"
+        message_dict["tool_call_id"] = message.tool_call_id
+
+        # tool message doesn't have name: https://platform.openai.com/docs/api-reference/chat/create#chat-create-messages
+        del message_dict["name"]
```

Comment:
Should we log a warning here? Will OpenAI reject this?

Reply:
OpenAI rejects. Think I should raise an error?

Reply:
changed to pass through unless it's none